python/sglang/srt/managers/router/manager.py

```diff
@@ -41,12 +41,16 @@ async def loop_for_forward(self):
                 self.send_to_detokenizer.send_pyobj(obj)
 
             # async sleep for receiving the subsequent request and avoiding cache miss
+            slept = False
             if len(out_pyobjs) != 0:
                 has_finished = any([obj.finished for obj in out_pyobjs])
                 if has_finished:
-                    await asyncio.sleep(self.extend_dependency_time)
+                    if self.extend_dependency_time > 0:
+                        slept = True
+                        await asyncio.sleep(self.extend_dependency_time)
 
-            await asyncio.sleep(0.0006)
+            if not slept:
+                await asyncio.sleep(0.0006)
```

Comment:
1. Did you find this give better performance?
2. @hnyls2002 please also take a look

Reply:
1. In my limited testing with low concurrency (1-2 concurrent requests with range 1-10 batches each) I did not find performance diff.

But this is unnecessary code/logic. There is no need to sleep for another 6ms (causing potential thread context switch) when it has slept for 30ms in earlier condition.